test/val/val_extensions_test.cpp

```diff
@@ -118,4 +118,70 @@ TEST_F(ValidateExtensionCapabilities, DeclCapabilityFailure) {
+
+TEST_P(ValidateAMDShaderBallotCapabilities, ExpectSuccess) {
+  const string preamble = R"(
+  OpCapability Shader
+  OpCapability Linkage
+  OpExtension "SPV_AMD_shader_ballot"
+  OpMemoryModel Logical GLSL450
+  %float = OpTypeFloat 32
+  %uint = OpTypeInt 32 0
+  %int = OpTypeInt 32 1
+  %scope = OpConstant %uint 3
+  %uint_const = OpConstant %uint 42
+  %int_const = OpConstant %uint 45
+  %float_const = OpConstant %float 3.5
+
+  %void = OpTypeVoid
+  %fn_ty = OpTypeFunction %void
+  %fn = OpFunction %void None %fn_ty
+  %entry = OpLabel
+  )";
+
+  const string assembly = preamble + GetParam() + "\nOpReturn OpFunctionEnd";
+
+  CompileSuccessfully(assembly.c_str());
+  ASSERT_EQ(SPV_SUCCESS, ValidateInstructions()) << getDiagnosticString();
+}
+
+INSTANTIATE_TEST_CASE_P(ExpectSuccess, ValidateAMDShaderBallotCapabilities,
+                        Values(
+                            // clang-format off
+  "%iadd_reduce = OpGroupIAddNonUniformAMD %uint %scope Reduce %uint_const",
+  "%iadd_iscan = OpGroupIAddNonUniformAMD %uint %scope InclusiveScan %uint_const",
+  "%iadd_escan = OpGroupIAddNonUniformAMD %uint %scope ExclusiveScan %uint_const",
+
+  "%fadd_reduce = OpGroupFAddNonUniformAMD %float %scope Reduce %float_const",
+  "%fadd_iscan = OpGroupFAddNonUniformAMD %float %scope InclusiveScan %float_const",
+  "%fadd_escan = OpGroupFAddNonUniformAMD %float %scope ExclusiveScan %float_const",
+
+  "%fmin_reduce = OpGroupFMinNonUniformAMD %float %scope Reduce %float_const",
+  "%fmin_iscan = OpGroupFMinNonUniformAMD %float %scope InclusiveScan %float_const",
+  "%fmin_escan = OpGroupFMinNonUniformAMD %float %scope ExclusiveScan %float_const",
+
+  "%umin_reduce = OpGroupUMinNonUniformAMD %uint %scope Reduce %uint_const",
+  "%umin_iscan = OpGroupUMinNonUniformAMD %uint %scope InclusiveScan %uint_const",
+  "%umin_escan = OpGroupUMinNonUniformAMD %uint %scope ExclusiveScan %uint_const",
+
+  "%smin_reduce = OpGroupUMinNonUniformAMD %int %scope Reduce %int_const",
+  "%smin_iscan = OpGroupUMinNonUniformAMD %int %scope InclusiveScan %int_const",
+  "%smin_escan = OpGroupUMinNonUniformAMD %int %scope ExclusiveScan %int_const",
+
+  "%fmax_reduce = OpGroupFMaxNonUniformAMD %float %scope Reduce %float_const",
+  "%fmax_iscan = OpGroupFMaxNonUniformAMD %float %scope InclusiveScan %float_const",
+  "%fmax_escan = OpGroupFMaxNonUniformAMD %float %scope ExclusiveScan %float_const",
+
+  "%umax_reduce = OpGroupUMaxNonUniformAMD %uint %scope Reduce %uint_const",
+  "%umax_iscan = OpGroupUMaxNonUniformAMD %uint %scope InclusiveScan %uint_const",
+  "%umax_escan = OpGroupUMaxNonUniformAMD %uint %scope ExclusiveScan %uint_const",
+
+  "%smax_reduce = OpGroupUMaxNonUniformAMD %int %scope Reduce %int_const",
+  "%smax_iscan = OpGroupUMaxNonUniformAMD %int %scope InclusiveScan %int_const",
+  "%smax_escan = OpGroupUMaxNonUniformAMD %int %scope ExclusiveScan %int_const"
+                            // clang-format on
+                            ));
+
```

Comment:
Consider testing that it fails without the extension.

Reply:
Fixed.